packages/typeorm/README.md

```diff
@@ -0,0 +1,145 @@
+# @loopback/typeorm
+
+This module enables TypeORM support in LoopBack.
+
+## Overview
+
+**NOTE**: This module is experimental and evolving.
+
+## Installation
+
+```sh
+npm install --save @loopback/typeorm
+```
+
+## Usage
+
+### Enabling TypeORM support
+
+Enabling TypeORM support requires the use of the `TypeOrmMixin` mixin and the
+`TypeOrmComponent` component. Import them from `@loopback/typeorm` and configure
+your app as shown below.
+
+```ts
+import {TypeOrmComponent, TypeOrmMixin} from '@loopback/typeorm';
+...
+export class AppaApplication extends BootMixin(
+  ServiceMixin(TypeOrmMixin(RestApplication)),
+) {
+  constructor(options: ApplicationConfig = {}) {
+    super(options);
+    ...
+    this.component(TypeOrmComponent);
+    ...
+  }
+}
+
+```
+
+### Creating Entities
+
+[Entities](https://typeorm.io/#/entities) are equivalent to Juggler's Models.
+Define the entities as usual and keep them in a directory named `entities`, or
+even `models` or any name, just make sure to load them correctly in the
+connections.
+
+```ts
+// src/entities/book.entity.ts
+import {Entity, Column, PrimaryColumn} from 'typeorm';
+@Entity()
+export class Photo {
+  @PrimaryColumn()
+  id: number;
+
+  @Column()
+  title: string;
+
+  @Column()
+  isPublished: boolean;
+}
+```
+
+### Creating Connections
+
+[Connections](https://typeorm.io/#/connection) are equivalent to Juggler
+datasources. They contain the connectivity and other details about the
+databases. Define the connection in files with a `.connection.ts` extension and
+keep them in the `connections` directory of the project.
+
+```ts
+// src/connections/sqlite.connection.ts
+import path from 'path';
+import {SqliteConnectionOptions} from '@loopback/typeorm';
+import {Book} from '../entities/';
+
+export const SQLLITE_CONNECTION: SqliteConnectionOptions = {
```

Comment:
Can we use `ConnectionOptions` so that `@loopback/typeorm` does not have to re-export internal TypeORM files like `typeorm/driver/mysql/MysqlConnectionOptions`?

Note that `ConnectionOptions` is a [discriminated union type](https://www.typescriptlang.org/docs/handbook/advanced-types.html#discriminated-unions), it includes `SqliteConnectionOptions` as the variant used when `type` is `sqlite`. As a result, the compiler is able to verify that all SQLite-related fields are correctly set, even when using `ConnectionOptions` as the type.

Example file:

```ts
import {ConnectionOptions} from 'typeorm';
export const config: ConnectionOptions = {
  type: 'sqlite',
};
```

Compiler output:

```
$  tsc x.ts
x.ts:2:14 - error TS2322: Type '{ type: "sqlite"; }' is not assignable to type 'ConnectionOptions'.
  Property 'database' is missing in type '{ type: "sqlite"; }' but required in type 'SqliteConnectionOptions'.

2 export const config: ConnectionOptions = {
               ~~~~~~

  node_modules/typeorm/driver/sqlite/SqliteConnectionOptions.d.ts:13:14
    13     readonly database: string;
                    ~~~~~~~~
    'database' is declared here.


Found 1 error.
```